spec/API_specification/array_api/array_object.py

```diff
@@ -110,28 +110,48 @@ def T(self: array) -> array:
 
     def __abs__(self: array, /) -> array:
         """
-        Calculates the absolute value for each element of an array instance (i.e., the element-wise result has the same magnitude as the respective element but has positive sign).
+        Calculates the absolute value for each element of an array instance.
+
+        For real-valued input arrays, the element-wise result has the same magnitude as the respective element in ``x`` but has positive sign.
 
         .. note::
            For signed integer data types, the absolute value of the minimum representable integer is implementation-dependent.
 
         **Special cases**
 
-        For floating-point operands, let ``self`` equal ``x``.
+        Let ``self`` equal ``x``.
+
+        For real-valued floating-point operands,
 
         -   If ``x_i`` is ``NaN``, the result is ``NaN``.
         -   If ``x_i`` is ``-0``, the result is ``+0``.
         -   If ``x_i`` is ``-infinity``, the result is ``+infinity``.
 
+        For complex floating-point operands, let ``a = real(x_i)``, ``b = imag(x_i)``, and
+
+        - If ``a`` is either ``+infinity`` or ``-infinity`` and ``b`` is any value (including ``NaN``), the result is ``+infinity``.
+        - If ``a`` is any value (including ``NaN``) and ``b`` is either ``+infinity`` or ``-infinity``, the result is ``+infinity``.
+        - If ``a`` is either ``+0`` or ``-0``, the result is equal to ``abs(b)``.
+        - If ``b`` is either ``+0`` or ``-0``, the result is equal to ``abs(a)``.
+        - If ``a`` is ``NaN`` and ``b`` is a finite number, the result is ``NaN``.
+        - If ``a`` is a finite number and ``b`` is ``NaN``, the result is ``NaN``.
+        - If ``a`` is ``NaN`` and ``b`` is ``NaN``, the result is ``NaN``.
+
+        .. note::
+           For complex floating-point operands, conforming implementations should take care to avoid undue overflow or underflow during intermediate stages of computation.
+
+        ..
+           TODO: once ``hypot`` is added to the specification, remove the special cases for complex floating-point operands and the note concerning guarding against undue overflow/underflow, and state that special cases must be handled as if implemented as ``hypot(real(x), imag(x))``.
```

Comment:
Checked in the rendered html that this isn't visible, so should be fine to leave that in here.